Create a context menu for switching card modes

jenweber opened the pull request:
Implemented a context menu. Closes #1287. Closes #1273.

<img width="703" alt="Screen Shot 2020-02-05 at 2 21 59 PM" src="https://user-images.githubusercontent.com/16627268/73886996-aca3b600-4838-11ea-8baf-4f349fb2c5d5.png">

Added:
- card-renderer-header component, which has the header and context menu
- styles for both the sticky header and the regularly positioned header
- when the header closes, the menu closes too
- clicking items in the menu takes you to other routes
- hover over menu items to see them highlighted
- checkbox next to the current mode
- tests
- percy snapshot

## Changed:

- instead of saving isSelected on the card record, I made it its own independent variable
- Moved the header markup/logic into its own component (card-renderer-header)
- Remove uses of mut

## How to review:

In each card mode, open the context menu. Scroll around, resize things, etc. to make sure it keeps looking good.

Please also have a look at the code/css itself. I'm open to suggestions for refactoring.

## Things I didn't do
This is a homemade menu that I don't think is accessible. I'd like to do that in a followup PR, unless someone already knows of an easy fix.

burieberry commented:
@jenweber actually I think the changes I made in my PR fix the issue I mentioned above, so you can ignore it for now

jenweber commented:
I'm struggling to position the menu on the schema page in a way that's responsive:

<img width="1344" alt="Screen Shot 2020-02-06 at 10 57 17 AM" src="https://user-images.githubusercontent.com/16627268/73954202-7feead80-48cf-11ea-8e2a-1caee993edb6.png">

These are the rules that define positioning:

```css

.context-menu-nav {
  position: absolute;
  z-index: var(--ch-actions-z-index);
  background-color: var(--ch-light);
  border-radius: 10px;
  box-shadow: 0 5px 10px rgba(0, 0, 0, 0.4);
  top: 42px;
  right: -130px;
  width: 200px;
  padding-bottom: 10px;
}

/* Whenever there's a fixed header, like in edit and schema, use alternate layout */

.edit .context-menu-nav {
  position: fixed;
  top: auto;
  margin-left: 405px;
  margin-top: -13px;
  right: auto;
}

.schema .context-menu-nav {
  position: fixed;
  margin-left: 0;
  top: 125px;
  right: var(--ch-right-edge-width);
}
```

On each page, the things to test are seeing if vertical scrolling affects the menu in a bad way, and whether resizing the browser window has a negative effect.

bagby commented:
One minor nitpick -- I think the whole line should be clickable, right now it's only the word that's clickable. This is easily fixed by setting the padding on the `<a>` and not the `<li>`.

<img width="666" alt="Screen Shot 2020-02-07 at 12 43 32 PM" src="https://user-images.githubusercontent.com/176297/74052310-a4b65400-49a7-11ea-915a-f476eee8a1e3.png">
